src/driver/sqlx_postgres.rs

```diff
@@ -215,10 +215,10 @@ impl From<PgQueryResult> for ExecResult {
     }
 }
 
-pub(crate) fn sqlx_query(stmt: &Statement) -> sqlx::query::Query<'_, Postgres, PgArguments> {
-    let mut query = sqlx::query(&stmt.sql);
+pub(crate) fn sqlx_query(stmt: &Statement) -> sqlx::query::Query<'_, Postgres, SqlxValues> {
     if let Some(values) = &stmt.values {
-        query = bind_query(query, values);
+        sqlx::query_with(&stmt.sql, SqlxValues(values.clone()))
+    } else {
+        sqlx::query_with(&stmt.sql, SqlxValues(Values(Vec::new())))
     }
-    query
 }
```

Comment:
Btw... I'm curious if we could avoid the `clone` by taking ownership of the `stmt: Statement`?

Reply:
I cannot take ownership of the `stmt: Statement`, because after `sqlx_query` we needed to send metrics:

```rust
#[instrument(level = "trace")]
pub async fn query_all(&self, stmt: Statement) -> Result<Vec<QueryResult>, DbErr> {
    debug_print!("{}", stmt);

    let query = sqlx_query(&stmt);
    if let Ok(conn) = &mut self.pool.acquire().await {
        crate::metric::metric!(self.metric_callback, &stmt, {
            match query.fetch_all(conn).await {
                Ok(rows) => Ok(rows.into_iter().map(|r| r.into()).collect()),
                Err(err) => Err(sqlx_error_to_query_err(err)),
            }
        })
    } else {
        Err(DbErr::Query(
            "Failed to acquire connection from pool.".to_owned(),
        ))
    }
}
```

Reply:
Right!

We can change `pub statement: &'a crate::Statement` into `pub statement: String` but then, worse still we need to `to_string` on `Statement` ahead of `metrics!`.
- https://github.com/SeaQL/sea-orm/blob/d532deb6363d997beae029a8983d64c7e9574af6/src/metric.rs#L8-L17

So, I'll support as-is - cloning `stmt: Statement`. Later in time, when we're optimizing for performance, We can move metric utility to compile time, i.e. provide a wrapper to existing connection. So that we don't have to clone `Statement` even when metric is turned off.

CC @tyt2y3

Reply:
Can't do it anyway, because:

```
pub fn query_with<'q, DB, A>(sql: &'q str, arguments: A) -> Query<'q, DB, A> ...
```

`query_with` doesnot take ownership of the query...

Reply:
I can make PR for add support `QueryOwned` into sqlx and this would help us.